vizro-ai/src/vizro_ai/plot/_response_models.py

```diff
@@ -26,12 +27,10 @@
 def _format_and_lint(code_string: str) -> str:
     # Tracking https://github.com/astral-sh/ruff/issues/659 for proper python API
     # Good example: https://github.com/astral-sh/ruff/issues/8401#issuecomment-1788806462
-    linted = subprocess.check_output(
-        ["ruff", "check", "--fix", "--exit-zero", "--silent", "--isolated", "-"], input=code_string, encoding="utf-8"
-    )
-    formatted = subprocess.check_output(
-        ["ruff", "format", "--silent", "--isolated", "-"], input=linted, encoding="utf-8"
-    )
+    # While we wait for the API, we can autoflake and black to process code strings.
+
+    removed_imports = autoflake.fix_code(code_string, remove_all_unused_imports=True)
+    formatted = format_str(removed_imports, mode=FileMode())
```

Comment:
```suggestion
    # Black doesn't yet have a Python API, so format_str might not work at some point in the future.
    # https://black.readthedocs.io/en/stable/faq.html#does-black-have-an-api
    formatted = black.format_str(removed_imports, mode=black.Mode())
```

* comment would be useful
* `FileMode` is a legacy name for `Mode`
* namespace `black` is a bit nicer than having two vaguely named global imports